libraries/AP_Terrain/TerrainIO.cpp

```diff
@@ -196,13 +189,9 @@ void AP_Terrain::open_file(void)
     }
 
     if (fd != -1) {
-        ::close(fd);
+        AP::FS().close(fd);
     }
-#if HAL_OS_POSIX_IO
-    fd = ::open(file_path, O_RDWR|O_CREAT|O_CLOEXEC, 0644);
-#else
-    fd = ::open(file_path, O_RDWR|O_CREAT|O_CLOEXEC);
-#endif
+    fd = AP::FS().open(file_path, O_RDWR|O_CREAT|O_CLOEXEC);
```

Comment:
I wonder why 'O_CLOEXEC' is here?
It is not handled and I think this flag is used for sockets and not for posix open.

Reply:
fixed (and it is relevant for files, but not on ChibiOS)